api/types/types.go

```diff
@@ -248,16 +248,23 @@ type Commit struct {
 // Info contains response of Engine API:
 // GET "/info"
 type Info struct {
-	ID                 string
-	Containers         int
-	ContainersRunning  int
-	ContainersPaused   int
-	ContainersStopped  int
-	Images             int
-	Driver             string
-	DriverStatus       [][2]string
-	SystemStatus       [][2]string `json:",omitempty"` // SystemStatus is only propagated by the Swarm standalone API
-	Plugins            PluginsInfo
+	ID                string
+	Containers        int
+	ContainersRunning int
+	ContainersPaused  int
+	ContainersStopped int
+	Images            int
+	Driver            string
+	DriverStatus      [][2]string
+	SystemStatus      [][2]string `json:",omitempty"` // SystemStatus is only propagated by the Swarm standalone API
+	Plugins           PluginsInfo
+	// Features contains the list of features indicating what features are
+	// enabled or disabled in the daemon configuration. The field contains
+	// a list of strings, formatted as `<feature-name>=<enabled|disabled>`.
+	//
+	// If a certain feature doesn't appear in this list then it's unset
+	// (i.e. neither true nor false).
+	Features           []string `json:"Features,omitempty"`
```

Comment:
Any reason this is a `[]string` with a micro-format instead of a `map[string]bool`?

Reply:
either `map[string]bool` or `map[string]string` (for status, `enabled`| `disabled` | ..)
but please don't use yet another string microformat within a JSON API

Reply:
Yeah, this was a 5 minute, quick implementation for discussion in the maintainers call; we did discuss using a map instead (possibly with custom marshaling to return a consistent order in the API response)

But then we also discussed if we really wanted this, because

- these options don't reflect the current state of these features (only if they're overridden)
- for the builder version; this is already exposed through other ways
- for the snapshotters, it'll be more useful to know what snapshotter is used (and what other snapshotters are available); both of those would effectively replace what "graphdriver" is currently showing, so the preference was to explore using that instead.

w.r.t. buildx; I wonder if that's information that should be part of the buildkit API 🤔 (/cc @crazy-max )